system/Database/OCI8/Forge.php

```diff
@@ -273,31 +273,14 @@ protected function _dropTable(string $table, bool $ifExists, bool $cascade)
         return $sql;
     }
 
-    protected function _processForeignKeys(string $table): string
+    protected function _processForeignKeys(string $table, array $allowActions = [], bool $onUpdate = false, bool $shortName = false): string
     {
-        $sql = '';
-
-        $allowActions = [
-            'CASCADE',
-            'SET NULL',
-            'NO ACTION',
-        ];
-
-        foreach ($this->foreignKeys as $fkey) {
-            $nameIndex            = $table . '_' . implode('_', $fkey['field']) . '_fk';
-            $nameIndexFilled      = $this->db->escapeIdentifiers($nameIndex);
-            $foreignKeyFilled     = implode(', ', $this->db->escapeIdentifiers($fkey['field']));
-            $referenceTableFilled = $this->db->escapeIdentifiers($this->db->DBPrefix . $fkey['referenceTable']);
-            $referenceFieldFilled = implode(', ', $this->db->escapeIdentifiers($fkey['referenceField']));
-
-            $formatSql = ",\n\tCONSTRAINT %s FOREIGN KEY (%s) REFERENCES %s(%s)";
-            $sql .= sprintf($formatSql, $nameIndexFilled, $foreignKeyFilled, $referenceTableFilled, $referenceFieldFilled);
-
-            if ($fkey['onDelete'] !== false && in_array($fkey['onDelete'], $allowActions, true)) {
-                $sql .= ' ON DELETE ' . $fkey['onDelete'];
-            }
+        $allowActions = ['CASCADE', 'SET NULL', 'NO ACTION'];
+
+        if (version_compare($this->db->getVersion(), '18.0.0', '<')) {
```

Comment:
Why `18.0.0 <`?

Reply:
In version 18 and on there isn't a 30byte limit on key name. It seems the version before 18 is 12 but i'm not totally sure. I just verified that in 18 the limit was gone.

Reply:
It looks like maybe even back to 12.2. But it depends on the compatibility setting.

https://docs.oracle.com/en/database/oracle/oracle-database/18/sqlrf/Database-Object-Names-and-Qualifiers.html#GUID-75337742-67FD-4EC0-985F-741C93D918DA

>The maximum length of identifier names depends on the value of the COMPATIBLE initialization parameter.
>If COMPATIBLE is set to a value of 12.2 or higher, then names must be from 1 to 128 bytes
>If COMPATIBLE is set to a value lower than 12.2, then names must be from 1 to 30 bytes long

Reply:
https://docs.oracle.com/en/database/oracle/oracle-database/12.2/sqlrf/Database-Object-Names-and-Qualifiers.html#GUID-75337742-67FD-4EC0-985F-741C93D918DA

Reply:
So use 12.2?